view/src/main/java/cmm/apps/esmorga/view/navigation/Navigation.kt

```diff
@@ -0,0 +1,35 @@
+package cmm.apps.esmorga.view.navigation
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+
+sealed class Navigation(val route: String) {
+    data object EventListScreen : Navigation("eventListScreen")
+    data object EventDetailScreen : Navigation("eventDetailScreen/{eventId}") {
+        fun createRoute(eventId: String) = "eventDetailScreen/$eventId"
+    }
+
+}
+
+const val GOOGLE_MAPS_PACKAGE = "com.google.android.apps.maps"
+
+fun openNavigationApp(context: Context, latitude: Double, longitude: Double) {
+    val uri = Uri.parse("geo:$latitude,$longitude")
+    if (isPackageAvailable(context, GOOGLE_MAPS_PACKAGE)) {
+        val mapIntent = Intent(Intent.ACTION_VIEW, uri)
+        mapIntent.setPackage(GOOGLE_MAPS_PACKAGE)
+        context.startActivity(mapIntent)
+    } else {
+        // do nothing
+    }
+}
+
+private fun isPackageAvailable(context: Context, appPackage: String) =
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/cmm-apps-android/EsmorgaAndroid/pull/5#issuecomment-2194728801>#267d9d</a><span> - 06/27/2024, 01:47 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'isPackageAvailable' function uses a try-catch block to check if the package is available. This can be optimized to avoid using exceptions for control flow, which is generally considered a bad practice due to performance overhead.
</div>
<div id="fix">
<b>Fix:</b> Use the 'PackageManager.getInstalledPackages' method to check if the package is available without relying on exceptions.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -29,7 +29,8 @@
 -private fun isPackageAvailable(context: Context, appPackage: String) =
 -    try {
 -        val appInfo = context.packageManager?.getApplicationInfo(appPackage, 0)
 -        appInfo != null && appInfo.enabled
 -    } catch (e: PackageManager.NameNotFoundException) {
 -        false
 -    }
 +private fun isPackageAvailable(context: Context, appPackage: String): Boolean {
 +    val packages = context.packageManager.getInstalledPackages(0)
 +    return packages.any { it.packageName == appPackage }
 +}
```
</div>
</details>
</div>
<br>


</div>